app/src/main/java/com/eveningoutpost/dexdrip/Home.java

```diff
@@ -385,6 +386,9 @@ protected void onCreate(Bundle savedInstanceState) {
         setVolumeControlStream(AudioManager.STREAM_MUSIC);
 
         checkedeula = checkEula();
+        if (!Pref.getString("units", "mgdl").equals("mgdl")) { // Only if the selected unit is mmol/L
+            handleUnitsChange(null, "mmol", null); // Trigger the correction of values (defaults) if needed based on the selected unit
```

Comment:
I don't think this should be here, we should only need to call handleUnitsChange() when the units are changed.

Reply:
I will create a video clip and add here showing why we need this.

Reply:
Install a previous version of xDrip on a test phone.  Change the unit to mmol/L.
Then, update xDrip to our current Nightly.  Go to the Persistent High Alert page.  You will see this as the threshold:

170 mmol/L

Change the unit to mg/dL and back to mmol/L.  You will see it change to 9.4 mmol/L.

Triggering the handle unit change right after xDrip start guarantees that will never happen.

Reply:
ok, but the onCreate method is not once on application start it is whenever the Home activity is created. It should be in `IdempotentMigrations.performAll()` but you need to check that there are no side effects as well because that `handleUnitsChange()` method does a few different things.

Reply:
Thanks a lot.  I will fix all of this in a few hours.

Thanks